src/System.Memory/src/System/Buffers/Binary/Reader.cs

```diff
@@ -98,17 +98,8 @@ public static ulong ReverseEndianness(ulong value)
         public static T ReadMachineEndian<T>(ReadOnlySpan<byte> buffer)
             where T : struct
         {
-#if IsPartialFacade
-            if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
-            {
-                throw new ArgumentException(SR.Format(SR.Argument_InvalidTypeWithPointersNotSupported, typeof(T)));
-            }
-#else
-            if (SpanHelpers.IsReferenceOrContainsReferences<T>())
-            {
-                ThrowHelper.ThrowArgumentException_InvalidTypeWithPointersNotSupported(typeof(T));
-            }
-#endif
+            SpanHelpers.ValidateTypeIsBlittable<T>();
```

Comment:
`RuntimeHelpers.IsReferenceOrContainsReferences<T>` needs to be called directly, not via wrapper method. The JIT won't be able to inline the wrapper method in all cases. This is introducing performance regression.

Reply:
Wouldn't the JIT be able to inline the helper method as well? If not would it be able to handle it if it didn't wrap the exception throwing as well?

Reply:
All generic methods wrapping other generic methods are problematic. They do not always inline well.

Reply:
In that case, I will revert this change. Is there any way to avoid having the #ifdef in multiple places (whenever I need to call IsReferenceOrContainsReferences) without a performance regression?

Reply:
You can put the whole thing in separate files. (I do not think it is an improvement.)

Reply:
OK @ahsonkhan thanks for trying to apply my feedback feel free to undo this part of the change and just do the other 2 pieces.